pkg/target_determinator.go

```diff
@@ -721,7 +721,16 @@ func doQueryDeps(context *Context, targets TargetsList) (*QueryResults, error) {
 		if err != nil {
 			return nil, fmt.Errorf("failed to parse label returned from query %s: %w", mt.Target, err)
 		}
-		if context.FilterIncompatibleTargets && !compatibleTargets[l] {
+		// Need to do this due to a change in bazel-gazelle & how equality between labels is determined.
+		// Likey happened in https://github.com/bazel-contrib/bazel-gazelle/pull/1911.
+		isCompatible := false
+		for k, _ := range compatibleTargets {
```

Comment:
This is potentially a pretty expensive loop... Can we instead do a single conversion of `compatibleTargets` from `map[label.Label]bool` to `map[string]bool`, and do `compatibleTargets[l.String()]` lookups, rather than an n^2 loop?